sirepo/pkcli/job_cmd.py

```diff
@@ -178,16 +188,54 @@ def _do_sbatch_status(msg, template):
 
 
 def _do_sequential_result(msg, template):
-    r = simulation_db.read_result(msg.runDir)
+    r = template_common.read_sequential_result(msg.runDir)
     # Read this first: https://github.com/radiasoft/sirepo/issues/2007
-    if (r.state != job.ERROR and hasattr(template, 'prepare_output_file')
-        and 'models' in msg.data
-    ):
-        template.prepare_output_file(msg.runDir, msg.data)
-        r = simulation_db.read_result(msg.runDir)
+    if (hasattr(template, 'prepare_sequential_output_file') and 'models' in msg.data):
+        template.prepare_sequential_output_file(msg.runDir, msg.data)
+        r = template_common.read_sequential_result(msg.runDir)
     return r
 
 
+def _on_do_compute_exit(success_exit, is_parallel, template, run_dir):
+    # locals() must be called before anything else so we only get the function
+    # arguments
+    kwargs = locals()
+
+    def _failure_exit():
+        a = _post_processing()
+        if not a:
+            a = _parse_python_errors()
+        return PKDict(state=job.ERROR, error=a)
+
+    def _parse_python_errors():
+        f = run_dir.join(template_common.RUN_LOG)
+        if f.exists():
+            m = re.search(
+                # TODO(e-carlin): I don't actually understand why this works.
+                # Discuss with someone who might.
+                r'Traceback.*\n(\s+.*\n)*(.*?)\n*',
```

Comment:
@robnagler I need some guidance with this. This regex "works" but I don't really understand why. I couldn't find any information on a good way to parse python tracebacks using a regular expression. [Here is an example](https://regex101.com/r/1c3D5z/1) of it in action. I think it may be more useful to have the last line (`AssertionError: a test error`) than the line I'm currently grabbing. Any suggestions for improvement?